### DISCUSSION TITLE
How to get the value of the ArrayVariable at the end of the last nonlinear iteration

### DISCUSSION THREAD

**Discussion by xieeeect** [idaholab/moose#27148] [Q&A General]
In each iteration of the steady-state calculation, I need the value of the array variable in the previous iteration, but I don't know how to get it. Thanks for the answer.   : )

**Comment by GiudGiud** [comment]
Hello

You can use the functor interfaces with a previous nonlinear state on the variable
This one https://mooseframework.inl.gov/docs/doxygen/moose/classMoose_1_1FunctorBase.html#a9ebde35cd92b43d6ccad6143f96bedec

and use it in a new auxkernel

Guillaume

**Reply by xieeeect** [reply]
Thanks for your help, I'm going to take a closer look.

**Comment by YaqiWang** [comment]
I am not sure what exactly you are looking for. Maybe you want to study this test https://github.com/idaholab/moose/blob/next/test/tests/executioners/fixed_point/nonlinear_fixed_point.i , which uses vector tagging and has a custom solve object with fixed point iterations.
